src/App.tsx

```diff
@@ -1,24 +1,221 @@
+    if (processingIds.includes(id) || id === 0) {
+      return true;
+    }
+
+    return false;
+  };
+
+  const ErrorTitle = handlerError(isError);
+
+  const onShowArrow = useMemo(() => {
+    return todos.every(todo => todo.completed)
+      ? setIsArrow(true)
+      : setIsArrow(false);
+  }, [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return filteredTodos(todos, filter);
+  }, [todos, filter]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setIsError(Error.UPLOAD);
+      });
+  }, []);
+
+  const onAdd = (todo: Todo) => {
+    const {
+      title, userId, completed,
+    } = todo;
+
+    setIsCreated(true);
+
+    return client.post<Todo>('/todos', { title, userId, completed })
+      .then(result => {
+        setTodos(prev => [...prev, result]);
+        dispatch({ type: ReducerType.RESET });
+      })
+      .catch(() => setIsError(Error.ADD))
+      .finally(() => setIsCreated(false));
+  };
+
+  const onDelete = (todoIdToDelete: number) => {
+    setProcessingIds(prev => [...prev, todoIdToDelete]);
+
+    removeTodo(todoIdToDelete)
+      .then(() => getTodos(USER_ID))
+      .then(setTodos)
+      .catch(() => setIsError(Error.REMOVE))
+      .then(() => setProcessingIds((prev) => {
+        return prev.filter(id => id !== todoIdToDelete);
+      }));
+  };
+
+  const onUpdate = (todoID: number, data: Partial<Todo>) => {
+    setProcessingIds(prev => [...prev, todoID]);
+
+    return updateTodo(todoID, data)
+      .then(() => setProcessingIds((prev) => {
+        return prev.filter(id => id !== todoID);
+      }))
+      .catch(() => setIsError(Error.UPDATE))
+      .then(() => getTodos(USER_ID))
+      .then(setTodos);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!tempTodo.title.trim()) {
+      return setIsError(Error.TITLE);
+    }
+
+    return onAdd(tempTodo);
+  };
+
+  const onClear = () => {
+    visibleTodos
+      .map(todo => (todo.completed ? onDelete(todo.id) : todo));
+  };
+
+  const handleArrow = () => {
+    if (!isArrow) {
+      todos.map(todo => (!todo.completed
+        ? onUpdate(todo.id, { completed: true })
+        : todo));
+
+      setIsArrow(true);
+    }
+
+    if (isArrow) {
+      todos.map(todo => (todo.completed
+        ? onUpdate(todo.id, { completed: false })
+        : todo));
+
+      setIsArrow(false);
+    }
+  };
+
+  // const handleSubmitRename = (
+  //   value: string,
+  //   idToRename: number,
+  //   e?: React.FormEvent,
+  // ) => {
+  //   if (e) {
+  //     e.preventDefault();
+  //   }
+
+  //   setProcessingIds(prev => [...prev, idToRename]);
+
+  //   return onUpdate(idToRename, { title: value })
+  //     .then(() => setProcessingIds((prev) => {
+  //       return prev.filter(id => id !== idToRename);
+  //     }));
+  // };
+
+  const dispatchTitle = (e: string) => {
+    dispatch({
+      type: ReducerType.TITLE,
+      newTitle: e,
+    });
+  };
```

Comment:
you name that it is an event but take event.target.value
mb let's take the event here & set it in newTitle: event.target.value
```suggestion
  const dispatchTitle = (e: Event...) => {
    dispatch({
      type: ReducerType.TITLE,
      newTitle: event.target.value,
    });
  };
```